tools/sotn_build/get_file_list.py

```diff
@@ -0,0 +1,311 @@
+    configPath: str = next(
+        (file for file in os.listdir(configDir) if configPattern.match(file)), None
+    )
+
+    if configPath:
+        configPath = os.path.realpath(os.path.join(configDir, configPath))
+        with open(configPath) as configFile:
+            config: dict = yaml.safe_load(configFile)
+
+        extraOptions: dict = {
+            "version": args.version,
+            "ovl": args.ovl,
+            "ignore_hidden": args.ignore_hidden,
+            "include_shared": args.include_shared,
+        }
+        config["options"].update(extraOptions)
+
+        return config
+    else:  # Raise an exception if no configPath was found
+        raise FileNotFoundError(
+            f"Could not find a {args.config_type} config for {args.version} {args.ovl} in {configDir}"
+        )
+
+
+def GetFileSegments(segments: list) -> list[list]:
+    """Returns a list of file segments from a list of segments"""
+    # Extracts segment information
+    segs: list = [seg for seg in segments if isinstance(seg, list)]
+    subsegs: list = [
+        sub
+        for seg in segments
+        if isinstance(seg, dict) and "subsegments" in seg
+        for sub in seg["subsegments"]
+    ]
+    # Converts docts that contain segment information into lists
+    segs.extend(
+        [seg.get("start", 0x0), seg["type"].replace("code", "c"), seg["name"]]
+        for seg in segments + subsegs
+        if isinstance(seg, dict) and "name" in seg and "type" in seg
+    )
+    # Only returns segment elements that are lists
+    return [seg for seg in segs + subsegs if isinstance(seg, list)]
+
+
+def NormalizeArgs(arg: list[str]) -> list[str]:
+    """Takes a list of items that are either a string of comma separated items or an individual item and normalizes it into a list of individual items"""
+    if arg:
+        return ",".join(arg).split(",")
+    else:
+        return []
+
+
+def GetAsmFiles(options: dict, segments: list[list]) -> set[str]:
+    """Constructs a list of asm files from a given segment list"""
+    asmPath: str = options["asm_path"].rstrip("/")
+    dataPath: str = os.path.join(options["asm_path"], "data").rstrip("/")
+
+    asmFiles: set = {
+        FILENAME_TEMPLATE.substitute(
+            path=asmPath, filename=seg[2] if len(seg) >= 3 else seg[1], ext="s"
+        )
+        for seg in segments
+        if len(seg) >= 2 and seg[1] in SEG_TYPES["asm"]
+    }
+
+    # A seg type is added with a leading . for each existing data seg type
+    if not options["ignore_hidden"]:
+        dataSegTypes = SEG_TYPES["data"] + tuple(
+            f".{type}" for type in SEG_TYPES["data"]
+        )
+
+    dataFiles: set = {
+        FILENAME_TEMPLATE.substitute(
+            path=dataPath,
+            # If a segment has 3 or more elements, then it has a name
+            # If it only has 1 or 2 elements, we use the memory address as the name
+            filename=seg[2] if len(seg) >= 3 else f"{seg[0]:X}",
```

Comment:
My main concern with this approach would be that it might be somewhat brittle to changes on the splat side

Noting that I am raising a problem and not a solution here 😄